src/reference/config/bindings/binding-asyncapi.md

```diff
@@ -0,0 +1,383 @@
+
+AsyncAPI spec definition filename mapped by a unique API spec identifier.
+
+#### options.tcp
+
+> `object`
+
+`client` specific `tcp` options.
+
+##### tpc.host
+
+> `string`
+
+Hostname or IP address.
+
+##### tcp.port
+
+> `integer` | `string` | `array` of `integer` | `array` of `string`
+
+Port number(s), including port number ranges.
+
+#### options.http
+
+> `object`
+
+`http` specific options.
+
+##### http.authorization
+
+> `object` as map of named properties
+
+Authorization by guard for the `HTTP/1.1` and `HTTP/2` protocols.
+
+```yaml
+authorization:
+  jwt:
+    credentials:
+      headers:
+        authorization: Bearer {credentials}
+```
+
+##### authorization.credentials
+
+> `object`
+
+Defines how to extract credentials from the HTTP request.
+
+##### credentials.cookies
+
+> `map` of `name: value` properties
+
+Named cookie value pattern with `{credentials}`.
+
+##### credentials.headers
+
+> `map` of `name: value` properties
+
+Named header value pattern with `{credentials}`, e.g. `"Bearer` `{credentials}"`.
+
+##### credentials.query
+
+> `map` of `name: value` properties
+
+Named query parameter value pattern with `{credentials}`.
+
+#### options.tls
+
+> `object`
+
+`tls` specific options.
+
+##### tls.version
+
+> `string`
+
+Protocol version.
+
+##### tls.keys
+
+> `array` of `string`
+
+A list of reference names for the Vault key.
+
+##### tls.trust
+
+> `array` of `string`
+
+A list of reference names for the Vault certificate.
+
+##### tls.signers
+
+> `array` of `string`
+
+A list of reference names for the Vault signer certificate.
+
+##### tls.trustcacerts
+
+> `boolean` | Default: `true` when trust is `null`
+
+Trust CA certificates.
+
+##### tls.sni\*
+
+> `array` of `string`
+
+A list of the Server Name Indications.
+
+##### tls.alpn
+
+> `array` of `string`
+
+Application protocols.
+
+##### tls.mutual
+
+> `enum` [ "required", "requested", "none" ] | Default: `"none"`
+
+Mutual authentication.
+
+### mqtt_kafka
+
+> `object`
+
+`mqtt_kafka` specific options.
+
+#### mqtt_kafka.channels
+
+> `object`
+
+AsyncAPI Kafka channels describing the necessary topics for the MQTT-Kafka mapping.
+
+```yaml
+mqtt_kafka:
+  channels:
+    sessions: mqttSessions
+    retained: mqttRetained
+    messages: mqttMessages
+```
+
+##### channels.sessions
+
+> `string`
+
+AsyncAPI Kafka sessions channel.
+
+```yaml
+sessions: mqttSessions
+```
+
+##### channels.retained
+
+> `string`
+
+AsyncAPI Kafka retained channel.
+
+```yaml
+retained: mqttRetained
+```
+
+##### channels.messages
+
+> `string`
+
+AsyncAPI Kafka messages channel.
+
+```yaml
+messages: mqttMessages
+```
+
+### routes[].when
+
+> `array` of `object`
+
+List of conditions to match this route when adapting `asyncapi` MQTT streams to `asyncapi` Kafka streams.
+Read more: [When a route matches](../../../concepts/config-intro.md#when-a-route-matches)
+
+#### when[].api-id
+
+> `string`
+
+AsyncAPI spec identifier that matches from `asyncapi` binding MQTT stream.
```

Comment:
Here is the callout to a "spec identifier".

Reply:
@vordimous LGTM. 👍